### ISSUE TITLE
Versioning

### ISSUE THREAD

**Issue by BloodyIron** [nextcloud/desktop#1102]
When you interact with files on the web interface, you get versioning of the files you work against. Edit a file? New version. Upload a modified copy of a file that already exists? New version.

When you use the desktop sync app, this functionality is broken. New versions aren't made when you modify files synced with desktop apps, and you also can't restore through the desktop app.

We need the desktop app to be extended to at least create new versions of files when modifications happen. Extending this further with the ability to interact with versions in the desktop UX would be even more valuable. And I'm not just saying Windows, I mean any OS the desktop client runs on.

I think this is one of the last items to really enable nextCloud to go toe to toe with things like OneDrive and others.

Please?

**Comment by Xaldimo** [comment]
This would be really helpful. We mainly use client sync and update documents daily. Having versioning on client sync would enable us to make restoring previous version easier. Currently we use rsync to create full daily backups but even then you can miss things.

**Comment by PhilippSchlesinger** [comment]
> Extending this further with the ability to interact with versions in the desktop UX would be even more valuable. 

It would be really great to be able to revert to a previous version in the desktop client ui (analogous to the nextcloud web version tab).

That functionality could be contained in the "more" menu (in yellow):
![grafik](https://user-images.githubusercontent.com/2735628/149923469-5a18c697-5f9c-4be4-b0ff-b5ee5957b8cf.png)

Until the development of this function, the could be simply a link to the web version tab of each file presenting a shortcut for the user to the version history of a file. 

In the end, having the version history of a file available in the file explorer's nextcloud context menu would be awesome.

**Comment by PhilippSchlesinger** [comment]
there is also a feature request regarding accessing versions in the contextual menu https://github.com/nextcloud/desktop/issues/1140

**Comment by BloodyIron** [comment]
Man it's 2022 and nextCloud still doesn't have this? Version history must just not be a priority at all... comedic, really. Tragic too.

**Comment by camilasan** [comment]
> In the end, having the version history of a file available in the file explorer's nextcloud context menu would be awesome.

The file history is now accessible from the client.

I will close this one in favor of #1140.

**Comment by BloodyIron** [comment]
@camilasan why are we closing the older issue for the newer one, instead of closing #1140 and having the work done here? So the work is not done?

**Comment by kaaax0815** [comment]
What is the status on this?

**Comment by billr73** [comment]
Is there any active work on this?

**Comment by BloodyIron** [comment]
Yeah with all the love given to the desktop client, this seems like an outstanding item that breaks versioning of files server-side etc.
